File: eBiblioteka/eBiblioteka.Core/Entities/Book.cs

namespace eBiblioteka.Core
{
    public class Book : BaseEntity
    {
        public string Title { get; set; } = null!;
        public string? ShortDescription { get; set; } = null!;
        public int? PublishingYear { get; set; }
        public int OpeningCount { get; set; }

        public int? CoverPhotoId { get; set; }
        public Photo? CoverPhoto { get; set; } = null!;

        public int AuthorID { get; set; }
        public Author Author { get; set; } = null!;

        public int? BookFileId { get; set; }
        public BookFile? BookFile { get; set; } 


        public ICollection<BookGenre> Genres { get; set; } = null!;
        public ICollection<Quote> Quotes { get; set; } = null!;
        public ICollection<UserBook> Readers { get; set; } = null!;
        public ICollection<Rating> UserRate { get; set; }=null!;

    }
}

File: eBiblioteka/UI/admin_ebiblioteka/lib/models/book.dart
import '../models/author.dart';
import '../models/photo.dart';
import 'package:json_annotation/json_annotation.dart';

part 'book.g.dart';

@JsonSerializable()
class Book {
  int? id;
  String? title;
  String? shortDescription;
  int? publishingYear;
  int? openingCount;
  bool? isActive;
  int? coverPhotoId;
  Photo? coverPhoto;
  int? authorID;
  Author? author;
  int? bookFileId;

  Book(
      this.id,
      this.title,
      this.shortDescription,
      this.publishingYear,
      this.openingCount,
      this.coverPhoto,
      this.coverPhotoId,
      this.author,
      this.authorID,
      this.bookFileId,
      this.isActive);

  factory Book.fromJson(Map<String, dynamic> json) => _$BookFromJson(json);

  Map<String, dynamic> toJson() => _$BookToJson(this);
}
